### PR TITLE
spec: disable await binding identifier within static block

### PR DESCRIPTION
**Pull request by JLHwung**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Implements https://github.com/tc39/proposal-class-static-block/pull/32
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

This PR forbids `await` binding within the static blocks. The logic of `arguments` check is extended for `await` binding.

Since `async (x = class { static { await; } }) => {}` is no longer valid, I also removed the expression scope creation for class static block, since `AwaitBindingIdentifierInStaticBlock ` would be thrown in this case.

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/12661"></a>

### CODE REVIEW

File: packages/babel-parser/test/fixtures/experimental/class-static-block/await-binding-in-function-in-static-block/input.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+var C;
+
+C = class { static { function f() { await } } };
+
+C = class { static { function f(await) {} } };
+
+C = class { static { function f(x = await) {} } };
+
+C = class { static { function f({ [await]: x }) {} } };
```
Comment (JLHwung) [inline packages/babel-parser/test/fixtures/experimental/class-static-block/await-binding-in-function-in-static-block/input.js:9]: Cases that `await` as binding will not throw.

File: packages/babel-parser/test/fixtures/experimental/class-static-block/await-binding-in-static-block/input.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+C = class { static { async ({ [await]: x}) => {} } };
+
+C = class { static { async ({ [await]: x}) } };
+
+async (x = class { static { await } }) => {};
+
+C = class { static { function await() {} } };
+
+C = class { static { await: 0 } };
```
Comment (JLHwung) [inline packages/babel-parser/test/fixtures/experimental/class-static-block/await-binding-in-static-block/input.js:27]: New behaviour: cases that `await` as binding throws.

**Review by nicolo-ribaudo — COMMENTED** [review]
Q: How is the context inside of a static block different from the one inside a static field?

**Comment by JLHwung** [comment]
> Q: How is the context inside of a static block different from the one inside a static field?

Compared to static field (initializer), the static block further blocks `await` as binding identifier which is allowed in `[~Await]` production. The motivation is to reserve `await` and "fail close" so that in the future we may support async static block, parallel to the async constructors.

**Review by nicolo-ribaudo — APPROVED** [review]

**Comment by JLHwung** [comment]
The CI error is fixed in #12716. I will rebase once that one is merged.

**Review by kaicataldo — APPROVED** [review]

**Comment by JLHwung** [comment]
Before merging this PR, please review #12716 first. 🙏

**Pull request merged**